acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_04.py

```diff
@@ -0,0 +1,155 @@
+        MyVars.test_report.set_test_failed()
+        MyVars.test_report.set_step_status(step, Status.FAILED, str(E))
+        print(Colors.FAIL + str(E) + Colors.ENDC)
+        return None
+
+    # 2. verify wallet was created in .indy/wallet
+    step = "Step02. Verify wallet was created in .indy/wallet"
+    try:
+        print(Colors.HEADER + "\n\t {0}\n".format(step) + Colors.ENDC)
+        work_dir = os.path.expanduser('~') + os.sep + ".indy"
+        wallet_path = work_dir + "/wallet/" + MyVars.wallet_name
+        result = os.path.exists(wallet_path)
+        if result:
+            MyVars.test_results['Step2'] = True
+            MyVars.test_report.set_step_status(step, Status.PASSED)
+            print("===PASSED===")
+    except IndyError as E:
+        MyVars.test_report.set_test_failed()
+        MyVars.test_report.set_step_status(step, Status.FAILED, str(E))
+        print(Colors.FAIL + str(E) + Colors.ENDC)
+
+    await asyncio.sleep(0)
+
+    # 3. create DID to check the new wallet work well.
+    step = "Step03. Create DID to check the new wallet work well"
+    print(Colors.HEADER + "\n\t {0}\n".format(step) + Colors.ENDC)
+    try:
+        # create and store did to check the new wallet work well.
+        (default_trustee_did, default_trustee_verkey) = await signus.create_and_store_my_did(
+            MyVars.wallet_handle, json.dumps({"seed": seed_default_trustee}))
+        if default_trustee_did:
+            MyVars.test_results['Step3'] = True
+            MyVars.test_report.set_step_status(step, Status.PASSED)
+            print("===PASSED===")
+    except IndyError as E:
+        MyVars.test_report.set_test_failed()
+        MyVars.test_report.set_step_status(step, Status.FAILED, str(E))
+        print(Colors.FAIL + str(E) + Colors.ENDC)
+
+    # ==================================================================================================================
+    #      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! End of test, run cleanup !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    # ==================================================================================================================
+    # 4. Close wallet and pool ------------------------------------------------------------------------------
+    step = "Step04. Close and delete the wallet and the pool ledger..."
+    print(Colors.HEADER + "\n\t {0}\n".format(step) + Colors.ENDC)
+    try:
+        await Common.clean_up_pool_and_wallet(MyVars.pool_name, MyVars.pool_handle, MyVars.wallet_name, MyVars.wallet_handle)
+        MyVars.test_results['Step4'] = True
+        MyVars.test_report.set_step_status(step, Status.PASSED)
+        print("===PASSED===")
+    except IndyError as E:
+        MyVars.test_report.set_test_failed()
+        MyVars.test_report.set_step_status(step, Status.FAILED, str(E))
+        print(Colors.FAIL + str(E) + Colors.ENDC)
+
+    await asyncio.sleep(0)
+    logger.info("Test Scenario 04 -> completed")
+
+
+def final_result():
```

Comment:
Can we make this methos Utility one to avoid code duplication?

Reply:
We restructured the source code. We put the step make_final_result into the TestBaseScenario class.
All test instances will inherit this test base class.